spec/classes/broker_spec.rb

```diff
@@ -2,172 +2,140 @@
 require 'shared_examples_param_validation'
 
 describe 'kafka::broker', type: :class do
-  let :facts do
-    {
-      osfamily: 'Debian',
-      os: { family: 'Debian' },
-      operatingsystem: 'Ubuntu',
-      operatingsystemrelease: '14.04',
-      lsbdistcodename: 'trusty',
-      architecture: 'amd64',
-      service_provider: 'upstart'
-    }
-  end
-  let :common_params do
-    {
-      config: {
-        'zookeeper.connect' => 'localhost:2181'
-      }
-    }
-  end
-
-  let :params do
-    common_params
-  end
-
-  it { is_expected.to contain_class('kafka::broker::install').that_comes_before('Class[kafka::broker::config]') }
-  it { is_expected.to contain_class('kafka::broker::config').that_comes_before('Class[kafka::broker::service]') }
-  it { is_expected.to contain_class('kafka::broker::service').that_comes_before('Class[kafka::broker]') }
-  it { is_expected.to contain_class('kafka::broker') }
-
-  context 'on Debian' do
-    describe 'kafka::broker::install' do
-      context 'defaults' do
-        it { is_expected.to contain_class('kafka') }
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
+        facts
       end
-    end
 
-    describe 'kafka::broker::config' do
-      context 'defaults' do
-        it { is_expected.to contain_file('/opt/kafka/config/server.properties') }
+      let :common_params do
```

Comment:
One pattern I'd suggest is (at the same level as `let(:facts)`:
```ruby
let(:params) do
  end
end
```

Then in a lower level you can use `let(:params) { super().merge(...) }`

Reply:
I am not sure how this is supposed to look like. ```:common_params``` is used multiple times and required by ```shared_examples_param_validation.rb```.

Reply:
It's not a blocker and in itself this is probably already an improvement.

Reply:
I would like to proceed without changing this.